lib/step.rb

```diff
@@ -9,10 +9,13 @@ class Step
     # Instantiates a Step object.
     # @param command [Array<String>] Command to execute and arguments.
     # @param env [Hash] Environment variables to set when running command.
-    def initialize(command, env:, verbose:)
-      @command = command
+    def initialize(command, env:, verbose:, named_args: nil, ignore_failures: false, repository: nil)
+      @named_args = [named_args].flatten.compact.map(&:to_s)
+      @command = command | @named_args
```

Comment:
Why the `|` here? I'm sure it's necessary, I just don't understand so a comment might be nice.

Reply:
The `test` method might be called in the following ways:

- `test "brew", "install", "foo", named_args: "foo"`
- `test "brew", "install", named_args: "foo"`

I only ever use the latter form here, but the former might, with future refactoring, be difficult to avoid (e.g. when the formula names are dynamically generated) and would likely produce rather confusing errors.

Reply:
> but the former might, with future refactoring, be difficult to avoid

If we don't need it for now: I'd say default to YAGNI and we can add `|` later if needed.

Reply:
Ok, done!